test/integration.nix

```diff
@@ -1,58 +1,92 @@
-{
-nixpkgs ? <nixpkgs>,
-pkgs ? import <nixpkgs> { inherit system; config = {}; },
-system ? builtins.currentSystem
-} @args:
+args@{ nixpkgs ? <nixpkgs>, ... }:
 
-import "${nixpkgs}/nixos/tests/make-test-python.nix" ({ pkgs, ...}: {
-  name = "agenix-integration";
+with (import "${nixpkgs}/lib");
 
-  nodes.system1 = { config, lib, ... }: {
+import "${nixpkgs}/nixos/tests/make-test-python.nix"
+  (
+    let
+      sshdConf = {
+        enable = true;
+        hostKeys = [{ type = "ed25519"; path = "/etc/ssh/ssh_host_ed25519_key"; }];
+      };
+    in
+    rec {
+      name = "agenix-integration";
+
+      nodes.system1 = { config, ... }: {
+        imports = [
+          ../modules/age.nix
+          ./install_ssh_host_keys.nix
+        ];
+
+        services.openssh = sshdConf;
 
-    imports = [
-      ../modules/age.nix
-      ./install_ssh_host_keys.nix
-    ];
+        age.secrets.passwordfile-user1.file = ../example/passwordfile-user1.age;
 
-    services.openssh.enable = true;
+        users = {
+          mutableUsers = false;
+
+          users = {
+            user1 = {
+              isNormalUser = true;
+              passwordFile = config.age.secrets.passwordfile-user1.path;
+            };
+          };
+        };
+      };
 
-    age.secrets.passwordfile-user1 = {
-      file = ../example/passwordfile-user1.age;
-    };
+      nodes.system2 = {
+        imports = [
+          ../modules/age.nix
+          ./install_ssh_host_keys.nix
+        ];
 
-    users = {
-      mutableUsers = false;
+        services.openssh = sshdConf;
 
-      users = {
-        user1 = {
-          isNormalUser = true;
-          passwordFile = config.age.secrets.passwordfile-user1.path;
+        age.secrets.ex1 = {
+          file = ../example/passwordfile-user1.age;
+          onChange = "echo bar > /tmp/foo";
```

Comment:
I think we should also test reloadUnits and restartUnits.

Reply:
good call. added.